frontend/src/app/modules/home/dashboard/dashboard.component.ts

```diff
@@ -207,4 +214,51 @@ export class DashboardComponent extends BaseComponent implements OnInit, OnDestr
                 this.toastNotificationService.error(error);
             });
     }
+
+    saveTilesOrder() {
+        this.spinnerService.show(true);
+
+        this.tiles = this.tiles.map((t, i) => ({ ...t, tileOrder: i + 1 }));
+        const orderingOfTiles = this.tiles.map(t => ({ id: t.id, tileOrder: t.tileOrder }));
+
+        this.tileService.setDashboardOrderForTiles(this.dashboard.id, orderingOfTiles)
+            .pipe(this.untilThis)
+            .subscribe(() => {
+                this.spinnerService.hide();
+                this.toastNotificationService.success('The Tiles order was successfully changed!');
+                this.isOrderChanged = false;
+            }, err => {
+                this.spinnerService.hide();
+                this.toastNotificationService.error(err);
+            });
+    }
+
+    drag(tile: Tile) {
+        this.draggableTile = tile;
+        this.canDrag = false;
+    }
+
+    drop(num: number) {
+        const removeIndex = this.tiles.indexOf(this.draggableTile);
+        this.tiles = this.tiles.filter(i => i.id !== this.draggableTile.id);
+
+        if (removeIndex < num) this.tiles.splice(num - 1, 0, this.draggableTile);
+        else this.tiles.splice(num, 0, this.draggableTile);
```

Comment:
Please refactor these lines. You have two almost similar lines,
And please try to get rid of splice

Reply:
Fixed